### ISSUE TITLE
Default localization isn't used for some fields

### ISSUE THREAD

**Issue by hckr** [mozilla/addons#430]
### Describe the problem and steps to reproduce it:
English (US) as default and only localization of an add-on.
Set descriptions of images, version details in English (US).
Go to add-on page listing (e.g. [my add-on](https://addons.mozilla.org/pl/firefox/addon/facebook-post-screenshot/)).
Select language other than English (US), e.g. Polish.

### What happened?
Gallery descriptions and version information details do not show up.

### What did you expect to happen?
Gallery descriptions and version information details should show up.
(Summary and add-on description show up no matter which language is selected.)

**Comment by jvillalobos** [comment]
I see the English strings when I visit your add-on using the Polish locale, like on the link you posted. Was it a temporary problem, or can you still reproduce it?

**Comment by hckr** [comment]
I'm sorry I misled you earlier. Problem reproduces only in Polish.

When I created the add-on page the default language was Polish and I mistakenly filled everything up and then removed Polish locale and chose English (US) as default and only one. But the system somehow sees that old data I removed.

Screenshot from the "Add-on developer hub":
![screenshot_20170628_005607](https://user-images.githubusercontent.com/1078650/27613604-a69ad4f4-5b9c-11e7-9581-4fd85b593742.png)

I managed to fix missing version descriptions (in "Version Information") by selecting Polish from the list above and then removing it (with "x" next to it). But descriptions for photos in gallery are still missing.

AMO language set to English (US):
![screenshot_20170628_004619](https://user-images.githubusercontent.com/1078650/27613372-80d0ad76-5b9b-11e7-862b-a8a6e527f515.png)
AMO language set to Polish:
![screenshot_20170628_004501](https://user-images.githubusercontent.com/1078650/27613490-10cb0660-5b9c-11e7-8039-9a27d8f76cba.png)
